app/src/main/java/ch/epfl/balelecbud/map/MapViewActivity.java

```diff
@@ -182,12 +187,28 @@ private void updateFriendLocation(User friend, Location location) {
         }
     }
 
+    private void displayPointsOfInterests() {
+        getAppDatabaseWrapper().query(new MyQuery(DatabaseWrapper.POINT_OF_INTEREST_PATH, new LinkedList<>()),
+                PointOfInterest.class).whenComplete((pointOfInterests, throwable) -> {
+                    for (PointOfInterest poi : pointOfInterests) {
+                        poiMarkers.put(poi, myMap.addMarker(new MyMarker.Builder()
+                            .location(poi.getLocation())
+                            .icon(icons.get(MarkerType.getMarkerType(poi.getType())))
+                            .title(poi.getName())));
+                    }
+        });
+    }
+
     private void setupMapIcons() {
-        icons = new HashMap<>();
         IconFactory iconFactory = IconFactory.getInstance(getApplicationContext());
-        Drawable iconDrawable = ContextCompat.getDrawable(getApplicationContext(), R.drawable.map);
-        Bitmap bitmap = ((BitmapDrawable) iconDrawable).getBitmap();
-        Icon icon = iconFactory.fromBitmap(bitmap);
-        icons.put(MarkerType.FRIEND, icon);
+        icons = new HashMap<>();
+        List<Integer> iconIds = Arrays.asList(R.drawable.friend_icon, R.drawable.poi_bar_icon,
+                R.drawable.poi_food_icon, R.drawable.poi_firstaid, R.drawable.poi_atm, R.drawable.poi_stage, R.drawable.map, R.drawable.map);
```

Comment:
You have two times `R.drawable.map` and for modularity sake you could put the drawable id in the `MarkerType` Enum

Reply:
Nice catch! Thanks 👍